pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -131,14 +178,28 @@ func (r *Reconciler) handleUpdate(ctx context.Context, crp *fleetv1beta1.Cluster
 	resourceSnapshotSpec := fleetv1beta1.ResourceSnapshotSpec{
 		SelectedResources: selectedResources,
 	}
-	_, err = r.getOrCreateClusterResourceSnapshot(ctx, crp, &resourceSnapshotSpec, int(revisionLimit))
+	latestResourceSnapshot, err := r.getOrCreateClusterResourceSnapshot(ctx, crp, &resourceSnapshotSpec, int(revisionLimit))
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	// update the status based on the latestPolicySnapshot status
-	// update the status based on the work
-	return ctrl.Result{}, nil
+	status, err := r.buildPlacementStatus(ctx, crp, latestSchedulingPolicySnapshot, latestResourceSnapshot)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Here we don't reset the crp status so that LastTransitionTime will be updated only if the new status of the condition
```

Comment:
Just a nit: why not pass in the old status for updating?

Reply:
I didn't fully get this